                                 Question-1

In [1]:
import numpy as np
import pandas as pd
import os
import re
import nltk
from nltk.stem import WordNetLemmatizer 
from nltk.tokenize import sent_tokenize, word_tokenize

In [2]:
data=pd.read_csv("/Users/user/Desktop/CUTE5/Data_Categories_TextClassification-1582933830366.csv") #Reading data

In [3]:
data.head()

Category                                        Description
0         1  meat manager Industrial Production manager Foo...
1         2  territory manager Sales manager Beverage and T...
2         1  senior vice president digital magazine act Chi...
3         1  supervisor technical service First-Line Superv...
4         1  registered nurse resource management Registere...

In [4]:
Text_Description = data['Description']

# Text cleaning and preprocessing

In [5]:
Text_Description_1 = Text_Description.tolist()
for i in range(len(Text_Description_1)):   
    Text_Description_1[i]=re.sub('[^A-Za-z]',' ',Text_Description_1[i])
    Text_Description_1[i]=re.sub('\s+',' ',Text_Description_1[i]) 
    #Replacing all the special characters other than alphabets with space using for loop
    #Also replacing more than one space with single space

In [6]:
Text_Description[1]

'territory manager Sales manager Beverage and Tobacco Product Manufacturing'

In [7]:
Text_Description_1[1]

'territory manager Sales manager Beverage and Tobacco Product Manufacturing'

In [8]:
Text_Description_1=[token.lower() for token in Text_Description_1]
Text_Description_1[1]     #replacing the uppercase letters into lowercase letters

'territory manager sales manager beverage and tobacco product manufacturing'

In [9]:
for i in range(len(Text_Description_1)):
    Text_Description_1[i] = word_tokenize(Text_Description_1[i])
Text_Description_1[1]     #tokenizing or splitting all words in a sentence

['territory',
 'manager',
 'sales',
 'manager',
 'beverage',
 'and',
 'tobacco',
 'product',
 'manufacturing']

In [10]:
from nltk.corpus import stopwords
stop = stopwords.words('english')
print(stop)

['i', 'me', 'my', 'myself', 'we', 'our', 'ours', 'ourselves', 'you', "you're", "you've", "you'll", "you'd", 'your', 'yours', 'yourself', 'yourselves', 'he', 'him', 'his', 'himself', 'she', "she's", 'her', 'hers', 'herself', 'it', "it's", 'its', 'itself', 'they', 'them', 'their', 'theirs', 'themselves', 'what', 'which', 'who', 'whom', 'this', 'that', "that'll", 'these', 'those', 'am', 'is', 'are', 'was', 'were', 'be', 'been', 'being', 'have', 'has', 'had', 'having', 'do', 'does', 'did', 'doing', 'a', 'an', 'the', 'and', 'but', 'if', 'or', 'because', 'as', 'until', 'while', 'of', 'at', 'by', 'for', 'with', 'about', 'against', 'between', 'into', 'through', 'during', 'before', 'after', 'above', 'below', 'to', 'from', 'up', 'down', 'in', 'out', 'on', 'off', 'over', 'under', 'again', 'further', 'then', 'once', 'here', 'there', 'when', 'where', 'why', 'how', 'all', 'any', 'both', 'each', 'few', 'more', 'most', 'other', 'some', 'such', 'no', 'nor', 'not', 'only', 'own', 'same', 'so', 'than', '

In [11]:
for i in range(len(Text_Description_1)):    #removing stowords present
    Text_Description_1[i] = [token for token in Text_Description_1[i] if token not in stop]

In [12]:
Text_Description_1[1]    #we can see There is stopwords are removed now

['territory',
 'manager',
 'sales',
 'manager',
 'beverage',
 'tobacco',
 'product',
 'manufacturing']

# lemmatizaton

In [13]:
lmtzr = nltk.stem.WordNetLemmatizer()
Text_Description_1[i] = [lmtzr.lemmatize(i) for i in Text_Description_1[i]]
print(Text_Description_1)

[['meat', 'manager', 'industrial', 'production', 'manager', 'food', 'beverage', 'stores'], ['territory', 'manager', 'sales', 'manager', 'beverage', 'tobacco', 'product', 'manufacturing'], ['senior', 'vice', 'president', 'digital', 'magazine', 'act', 'chief', 'executives', 'professional', 'scientific', 'technical', 'services'], ['supervisor', 'technical', 'service', 'first', 'line', 'supervisors', 'mechanics', 'installers', 'repairers', 'broadcasting', 'except', 'internet'], ['registered', 'nurse', 'resource', 'management', 'registered', 'nurse', 'hospitals'], ['registered', 'nurse', 'speciality', 'registered', 'nurse', 'administration', 'human', 'resource', 'programs'], ['territory', 'manager', 'iii', 'sales', 'manager', 'beverage', 'tobacco', 'product', 'manufacturing'], ['case', 'dock', 'laborer', 'laborers', 'freight', 'stock', 'material', 'movers', 'hand', 'merchant', 'wholesalers', 'nondurable', 'goods'], ['machine', 'operator', 'ww', 'packaging', 'filling', 'machine', 'operators'

In [14]:
for i in range(len(Text_Description_1)):
    Text_Description_1[i]=' '.join(Text_Description_1[i])

In [15]:
Text_Description_1[1]

'territory manager sales manager beverage tobacco product manufacturing'

# Splitting data into train and test

In [16]:
from sklearn.model_selection import train_test_split 
X_train, X_test, y_train, y_test=train_test_split(Text_Description_1,data['Category'],test_size=0.3,random_state=1234)

In [17]:
#checking the distribution of target classes in train,here 3 classes 
y_train.value_counts()

1    1380
2     613
3      95
Name: Category, dtype: int64

In [18]:
#checking the distribution of target classes in test,in test also there are same no of classes as in train
y_test.value_counts()

1    572
2    284
3     39
Name: Category, dtype: int64

# Implementation of vector space models

In [19]:
from sklearn.feature_extraction.text import CountVectorizer,TfidfVectorizer

In [20]:
tfidf_transformer = TfidfVectorizer(ngram_range=(1,1),stop_words='english')
X_train_tfidf = tfidf_transformer.fit_transform(X_train)
X_train_tfidf.shape

(2088, 652)

In [21]:
X_train[0]

'territory manager ii sales manager beverage tobacco product manufacturing'

In [22]:
X_test_tfidf = tfidf_transformer.transform(X_test)
X_test_tfidf.shape

(895, 652)

# Development of classification models

# Implementing Naive Bayes Classification Model

In [23]:
from sklearn.naive_bayes import MultinomialNB
NB = MultinomialNB().fit(X_train_tfidf, y_train) #NB is the classifier
pred_train = NB.predict(X_train_tfidf) #predictions on train data

from sklearn.metrics import confusion_matrix
prediction=confusion_matrix(y_train, pred_train) #confusion matrix for train predictions

In [24]:
print("Train_Confusion Matrix:\n",prediction,"\n")

from sklearn.metrics import recall_score, precision_score, accuracy_score

acc = accuracy_score(y_train, pred_train)
rec = recall_score(y_train, pred_train, labels=[1,2,3],average='micro')
prec = precision_score(y_train, pred_train, labels=[1,2,3],average='micro')

print("Results of Naive Bayes on train data:","\nAccuracy:",acc, "\nRecall:",rec, "\nPrecision:",prec)

Train_Confusion Matrix:
 [[1347   32    1]
 [   7  605    1]
 [   3   28   64]] 

Results of Naive Bayes on train data: 
Accuracy: 0.9655172413793104 
Recall: 0.9655172413793104 
Precision: 0.9655172413793104


In [25]:
pred = NB.predict(X_test_tfidf) #predicting on test data
prediction = confusion_matrix(y_test,pred) ##confusion matrix for test predictions

# Evaluation of model

In [26]:
print("Test_Confusion matrix:\n", prediction,"\n")
acc = accuracy_score(y_test,pred)
rec = recall_score(y_test,pred,labels=[1,2,3],average='micro')
prec=precision_score(y_test,pred,labels=[1,2,3],average='micro')
print("Results of Naive Bayes on test data:","\nAcc: ",acc,"\nRec:", rec, "\nPrec:", prec)

Test_Confusion matrix:
 [[551  20   1]
 [  3 281   0]
 [  5  18  16]] 

Results of Naive Bayes on test data: 
Acc:  0.9474860335195531 
Rec: 0.9474860335195531 
Prec: 0.9474860335195531


# Implementation of Logistic Regression model

In [27]:
from sklearn import linear_model

logreg_train = linear_model.LogisticRegression()
lr_clf = logreg_train.fit(X_train_tfidf, y_train) #fit logreg model on train data

pred_lr_train=lr_clf.predict(X_train_tfidf) # predict on train data
pred_lr_test=lr_clf.predict(X_test_tfidf) # predict on test data


print("Train_Confusion matrix:\n", confusion_matrix(y_train,pred_lr_train))
print("Test_Confusion matrix:\n",confusion_matrix(y_test,pred_lr_test))

Train_Confusion matrix:
 [[1379    1    0]
 [  11  596    6]
 [   4    9   82]]
Test_Confusion matrix:
 [[570   1   1]
 [  6 276   2]
 [  0   8  31]]


/opt/anaconda3/lib/python3.7/site-packages/sklearn/linear_model/logistic.py:432: FutureWarning: Default solver will be changed to 'lbfgs' in 0.22. Specify a solver to silence this warning.
  FutureWarning)
/opt/anaconda3/lib/python3.7/site-packages/sklearn/linear_model/logistic.py:469: FutureWarning: Default multi_class will be changed to 'auto' in 0.22. Specify the multi_class option to silence this warning.
  "this warning.", FutureWarning)


# Evaluation of model

In [28]:
print("Results of Logistic regression on train data")
acc=accuracy_score(y_train,pred_lr_train)
rec=recall_score(y_train,pred_lr_train,labels=[1,2,3],average='micro')
prec=precision_score(y_train,pred_lr_train,labels=[1,2,3],average='micro')
print("\nAcc: ",acc,"\nRec:", rec, "\nPrec:", prec)

Results of Logistic regression on train data

Acc:  0.9851532567049809 
Rec: 0.9851532567049809 
Prec: 0.9851532567049809


In [29]:
print("Results of Logistic regression on test data")
acc=accuracy_score(y_test,pred_lr_test)
rec=recall_score(y_test,pred_lr_test,labels=[1,2,3],average='micro')
prec=precision_score(y_test,pred_lr_test,labels=[1,2,3],average='micro')
print("\nAcc: ",acc,"\nRec:", rec, "\nPrec:", prec)

Results of Logistic regression on test data

Acc:  0.9798882681564246 
Rec: 0.9798882681564246 
Prec: 0.9798882681564246


# count vectorizer

In [30]:
from sklearn.feature_extraction.text import CountVectorizer
cv=CountVectorizer(lowercase=True,
                   strip_accents='unicode',decode_error='ignore')

tdm_train = cv.fit_transform(X_train)
Mat = tdm_train.todense()
Mat
Mat.shape

(2088, 662)

In [31]:
tdm_test = cv.transform(X_test)
Mat_test = tdm_test.todense()
Mat_test.shape

(895, 662)

# Logistic Regression model

In [32]:
from sklearn.linear_model import LogisticRegression
import warnings
warnings.filterwarnings("ignore") #just to ignore any warnings

#Training the model
logreg = LogisticRegression()
lr_clf = logreg.fit(tdm_train, y_train)


#Predicting on train data
train_pred = lr_clf.predict(tdm_train)


#Predicting on test data
test_pred=lr_clf.predict(tdm_test)

In [33]:
print("Train_Confusion Matrix: \n", confusion_matrix(y_train,train_pred))
print("Test_Confusion Matrix: \n", confusion_matrix(y_test,test_pred))

Train_Confusion Matrix: 
 [[1380    0    0]
 [   2  605    6]
 [   1    8   86]]
Test_Confusion Matrix: 
 [[568   3   1]
 [  4 279   1]
 [  0   9  30]]


# Evaluation of model

In [34]:
from sklearn.metrics import recall_score, precision_score, accuracy_score

acc = accuracy_score(y_train, train_pred)
rec = recall_score(y_train, train_pred,labels=[1,2,3], average='micro')
prec = precision_score(y_train, train_pred,labels=[1,2,3], average='micro')

print("Results of logistic regression on train data:","\nAccuracy:",acc, "\nRecall:",rec, "\nPrecision:",prec)

Results of logistic regression on train data: 
Accuracy: 0.9918582375478927 
Recall: 0.9918582375478927 
Precision: 0.9918582375478927


In [35]:
acc = accuracy_score(y_test, test_pred)
rec = recall_score(y_test, test_pred,labels=[1,2,3], average='micro')
prec = precision_score(y_test, test_pred,labels=[1,2,3], average='micro')

print("Results of logistic regression on test data:","\nAccuracy:",acc, "\nRecall:",rec, "\nPrecision:",prec)

Results of logistic regression on test data: 
Accuracy: 0.9798882681564246 
Recall: 0.9798882681564246 
Precision: 0.9798882681564246


# Naive Bayes classifier Model

In [36]:
from sklearn.naive_bayes import MultinomialNB
NB1 = MultinomialNB().fit(tdm_train, y_train) #NB is the classifier
pred_train1 = NB1.predict(tdm_train) #predictions on train data

from sklearn.metrics import confusion_matrix
prediction1=confusion_matrix(y_train, pred_train1) #confusion matrix for train predictions

In [37]:
print("Train_Confusion Matrix:\n",prediction,"\n")

from sklearn.metrics import recall_score, precision_score, accuracy_score

acc = accuracy_score(y_train, pred_train1)
rec = recall_score(y_train, pred_train1, labels=[1,2,3],average='micro')
prec = precision_score(y_train, pred_train1, labels=[1,2,3],average='micro')

print("Results of Naive Bayes on train data:","\nAccuracy:",acc, "\nRecall:",rec, "\nPrecision:",prec)

Train_Confusion Matrix:
 [[551  20   1]
 [  3 281   0]
 [  5  18  16]] 

Results of Naive Bayes on train data: 
Accuracy: 0.935823754789272 
Recall: 0.935823754789272 
Precision: 0.935823754789272


In [38]:
pred1 = NB1.predict(tdm_test) #predicting on test data
prediction = confusion_matrix(y_test,pred1) ##confusion matrix for test predictions

In [39]:
print("Test_Confusion matrix:\n", prediction,"\n")
acc = accuracy_score(y_test,pred1)
rec = recall_score(y_test,pred1,labels=[1,2,3],average='micro')
prec=precision_score(y_test,pred1,labels=[1,2,3],average='micro')
print("Results of Naive Bayes on test data:","\nAcc: ",acc,"\nRec:", rec, "\nPrec:", prec)

Test_Confusion matrix:
 [[522  25  25]
 [  0 279   5]
 [  0   9  30]] 

Results of Naive Bayes on test data: 
Acc:  0.9284916201117318 
Rec: 0.9284916201117318 
Prec: 0.9284916201117318


# Development of Vector Space Models (experiment on multiple parameters)

changing ngram_range to (1,2) and implementing model

In [40]:
tfidf_transformer = TfidfVectorizer(ngram_range=(1,2),stop_words='english')
X_train_tfidf = tfidf_transformer.fit_transform(X_train)
X_train_tfidf.shape
X_test_tfidf = tfidf_transformer.transform(X_test)
X_test_tfidf.shape

(895, 2345)

In [41]:
from sklearn.naive_bayes import MultinomialNB
NB = MultinomialNB().fit(X_train_tfidf, y_train) #NB is the classifier
pred_train = NB.predict(X_train_tfidf) #predictions on train data

from sklearn.metrics import confusion_matrix
prediction=confusion_matrix(y_train, pred_train) #confusion matrix for train predictions

In [42]:
print("Train_Confusion Matrix:\n",prediction,"\n")

from sklearn.metrics import recall_score, precision_score, accuracy_score

acc = accuracy_score(y_train, pred_train)
rec = recall_score(y_train, pred_train, labels=[1,2,3],average='micro')
prec = precision_score(y_train, pred_train, labels=[1,2,3],average='micro')

print("Results of Naive Bayes on train data:","\nAccuracy:",acc, "\nRecall:",rec, "\nPrecision:",prec)

Train_Confusion Matrix:
 [[1360   20    0]
 [   4  609    0]
 [   4   26   65]] 

Results of Naive Bayes on train data: 
Accuracy: 0.9741379310344828 
Recall: 0.9741379310344828 
Precision: 0.9741379310344828


In [43]:
pred = NB.predict(X_test_tfidf) #predicting on test data
prediction = confusion_matrix(y_test,pred) ##confusion matrix for test predictions

In [44]:
print("Test_Confusion matrix:\n", prediction,"\n")
acc = accuracy_score(y_test,pred)
rec = recall_score(y_test,pred,labels=[1,2,3],average='micro')
prec=precision_score(y_test,pred,labels=[1,2,3],average='micro')
print("Results of Naive Bayes on test data:","\nAcc: ",acc,"\nRec:", rec, "\nPrec:", prec)

Test_Confusion matrix:
 [[559  12   1]
 [  3 281   0]
 [  5  18  16]] 

Results of Naive Bayes on test data: 
Acc:  0.9564245810055866 
Rec: 0.9564245810055866 
Prec: 0.9564245810055866


#taking ngrams=(1,3)


In [45]:
tfidf_transformer = TfidfVectorizer(ngram_range=(1,3),stop_words='english')
X_train_tfidf = tfidf_transformer.fit_transform(X_train)
X_train_tfidf.shape
X_test_tfidf = tfidf_transformer.transform(X_test)
X_test_tfidf.shape

(895, 4408)

In [46]:
from sklearn.naive_bayes import MultinomialNB
NB = MultinomialNB().fit(X_train_tfidf, y_train) #NB is the classifier
pred_train = NB.predict(X_train_tfidf) #predictions on train data

from sklearn.metrics import confusion_matrix
prediction=confusion_matrix(y_train, pred_train) #confusion matrix for train predictions

In [47]:
print("Train_Confusion Matrix:\n",prediction,"\n")

from sklearn.metrics import recall_score, precision_score, accuracy_score

acc = accuracy_score(y_train, pred_train)
rec = recall_score(y_train, pred_train, labels=[1,2,3],average='micro')
prec = precision_score(y_train, pred_train, labels=[1,2,3],average='micro')

print("Results of Naive Bayes on train data:","\nAccuracy:",acc, "\nRecall:",rec, "\nPrecision:",prec)

Train_Confusion Matrix:
 [[1357   23    0]
 [   4  609    0]
 [   4   24   67]] 

Results of Naive Bayes on train data: 
Accuracy: 0.9736590038314177 
Recall: 0.9736590038314177 
Precision: 0.9736590038314177


In [48]:
pred = NB.predict(X_test_tfidf) #predicting on test data
prediction = confusion_matrix(y_test,pred) ##confusion matrix for test predictions

In [49]:
print("Test_Confusion matrix:\n", prediction,"\n")
acc = accuracy_score(y_test,pred)
rec = recall_score(y_test,pred,labels=[1,2,3],average='micro')
prec=precision_score(y_test,pred,labels=[1,2,3],average='micro')
print("Results of Naive Bayes on test data:","\nAcc: ",acc,"\nRec:", rec, "\nPrec:", prec)

Test_Confusion matrix:
 [[556  15   1]
 [  3 281   0]
 [  7  16  16]] 

Results of Naive Bayes on test data: 
Acc:  0.9530726256983241 
Rec: 0.9530726256983241 
Prec: 0.9530726256983241


# count vectoriser
#ngram_range=(1,2)

In [50]:
from sklearn.feature_extraction.text import CountVectorizer
cv=CountVectorizer(lowercase=True,ngram_range=(1,2),
                   strip_accents='unicode',decode_error='ignore')

tdm_train = cv.fit_transform(X_train)
Mat = tdm_train.todense()
Mat
Mat.shape

(2088, 2378)

In [51]:
tdm_test = cv.transform(X_test)
Mat_test = tdm_test.todense()
Mat_test.shape

(895, 2378)

In [52]:
from sklearn.linear_model import LogisticRegression
import warnings
warnings.filterwarnings("ignore") #just to ignore any warnings

#Training the model
logreg = LogisticRegression()
lr_clf = logreg.fit(tdm_train, y_train)


#Predicting on train data
train_pred = lr_clf.predict(tdm_train)


#Predicting on test data
test_pred=lr_clf.predict(tdm_test)

In [53]:
from sklearn.metrics import recall_score, precision_score, accuracy_score

acc = accuracy_score(y_train, train_pred)
rec = recall_score(y_train, train_pred,labels=[1,2,3], average='micro')
prec = precision_score(y_train, train_pred,labels=[1,2,3], average='micro')

print("Results of logistic regression on train data:","\nAccuracy:",acc, "\nRecall:",rec, "\nPrecision:",prec)

Results of logistic regression on train data: 
Accuracy: 0.9923371647509579 
Recall: 0.9923371647509579 
Precision: 0.9923371647509579


In [54]:
acc = accuracy_score(y_test, test_pred)
rec = recall_score(y_test, test_pred,labels=[1,2,3], average='micro')
prec = precision_score(y_test, test_pred,labels=[1,2,3], average='micro')

print("Results of logistic regression on test data:","\nAccuracy:",acc, "\nRecall:",rec, "\nPrecision:",prec)

Results of logistic regression on test data: 
Accuracy: 0.9810055865921787 
Recall: 0.9810055865921787 
Precision: 0.9810055865921787


In [55]:
#count vectoriser
#ngram_range=(1,3)

In [56]:
from sklearn.feature_extraction.text import CountVectorizer
cv=CountVectorizer(lowercase=True,ngram_range=(2,2),
                   strip_accents='unicode',decode_error='ignore')

tdm_train = cv.fit_transform(X_train)
Mat = tdm_train.todense()
Mat
Mat.shape

(2088, 1716)

In [57]:
tdm_test = cv.transform(X_test)
Mat_test = tdm_test.todense()
Mat_test.shape

(895, 1716)

In [58]:
from sklearn.linear_model import LogisticRegression
import warnings
warnings.filterwarnings("ignore") #just to ignore any warnings

#Training the model
logreg = LogisticRegression()
lr_clf = logreg.fit(tdm_train, y_train)


#Predicting on train data
train_pred = lr_clf.predict(tdm_train)


#Predicting on test data
test_pred=lr_clf.predict(tdm_test)

In [59]:
from sklearn.metrics import recall_score, precision_score, accuracy_score

acc = accuracy_score(y_train, train_pred)
rec = recall_score(y_train, train_pred,labels=[1,2,3], average='micro')
prec = precision_score(y_train, train_pred,labels=[1,2,3], average='micro')

print("Results of logistic regression on train data:","\nAccuracy:",acc, "\nRecall:",rec, "\nPrecision:",prec)

Results of logistic regression on train data: 
Accuracy: 0.9913793103448276 
Recall: 0.9913793103448276 
Precision: 0.9913793103448276


In [60]:
acc = accuracy_score(y_test, test_pred)
rec = recall_score(y_test, test_pred,labels=[1,2,3], average='micro')
prec = precision_score(y_test, test_pred,labels=[1,2,3], average='micro')

print("Results of logistic regression on test data:","\nAccuracy:",acc, "\nRecall:",rec, "\nPrecision:",prec)

Results of logistic regression on test data: 
Accuracy: 0.9787709497206704 
Recall: 0.9787709497206704 
Precision: 0.9787709497206704


#logistic regression on count vectoriser with ngrams=(1,2) gives very good accuracy,precision,recall scores of 99.2% on train data
#and 98.1% on test data


#  word frequency when we change ngram range to bigrams and trigrams

In [62]:
from sklearn.feature_extraction.text import CountVectorizer
word_vectorizer = CountVectorizer(ngram_range=(1,2), analyzer='word')
sparse_matrix = word_vectorizer.fit_transform(data['Description'])
frequencies = sum(sparse_matrix).toarray()[0]
pd.DataFrame(frequencies, index=word_vectorizer.get_feature_names(), columns=['frequency'])

frequency
abuse                       1
abuse social                1
account                   124
account associate           2
account collectors          1
...                       ...
xray radiologic             9
yard                       50
yard driver                18
yard jockey                 1
yard merchant              31

[2816 rows x 1 columns]

In [63]:
from sklearn.feature_extraction.text import CountVectorizer
word_vectorizer = CountVectorizer(ngram_range=(1,1), analyzer='word')
sparse_matrix = word_vectorizer.fit_transform(data['Description'])
frequencies = sum(sparse_matrix).toarray()[0]
pd.DataFrame(frequencies, index=word_vectorizer.get_feature_names(), columns=['frequency'])

frequency
abuse                1
account            124
accounts             6
acquisition          2
act                 11
...                ...
writers             10
wrker                2
ww                  10
xray                 9
yard                50

[733 rows x 1 columns]

# Finalisation of model

Logistic Regression model on count vectoriser with 99%accuracy,99%recall,99%precision on train data
and 98%accuracy,98%recall,98%precision on test data

# Choosing evaluation metric

here we want to predict which job description should fall in which category
so we want true predictions so i choose Accuracy here
Accuracy=((Σ True positive predictions + Σ True negative predictions)/Σ Total predictions)